app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -118,7 +120,7 @@ class ConversationViewModel @Inject constructor(
     private fun fetchMessages() = viewModelScope.launch {
         getMessages(conversationId).combine(observeMemberDetails(conversationId)) { messages, members ->
             messages.toUIMessages(members)
-        }.collect { uiMessages ->
+        }.flowOn(dispatchers.default()).collect { uiMessages ->
```

Comment:
Wouldn't this do exactly the same than the previous code as default as fetch messages is triggered by the main thread too? shouldn't we have to change the original call thread to `IO`? i.e `viewModelScope.launch(dispatchers.io()) {}` ??

Reply:
@gongracr [`Dispatchers.Default`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-default.html) is different from [`Dispatchers.Main`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-main.html).

## Default
A small thread pool for computation. Size: number of cores, but at least = 2.

## Main
Single threaded dispatcher. Uses the main/UI thread.

Reply:
Nice recap there and sorry for the confusion. Having said that, I think it would still be interesting to launch the `getMessages` call with the `IO` dispatcher as using `default` might not be enough for cases of peak activity (init syncing and loading of messages)